cassiopeia/src/main/java/edu/asu/diging/gilesecosystem/cassiopeia/web/EditPropertiesController.java

```diff
@@ -36,12 +39,49 @@
     private ISystemMessageHandler messageHandler;
 
     Map<String, String> ocrTypeMap = new HashMap<>();
+    Map<String, String> langTypeMap = new HashMap<>();
+    String defaultLang = new String();
+
 
     @InitBinder
     protected void initBinder(HttpServletRequest request, ServletRequestDataBinder binder, WebDataBinder validateBinder) {
         validateBinder.addValidators(new SystemConfigValidator());
         ocrTypeMap.put(Properties.OCR_PLAINTEXT, propertyManager.getProperty(Properties.OCR_PLAINTEXT));
         ocrTypeMap.put(Properties.OCR_HOCR, propertyManager.getProperty(Properties.OCR_HOCR));
+        
+        String[] langs = getTessLangs();
+        for(int i=1;i<langs.length;i++) {
+        	langTypeMap.put(langs[i],langs[i]);
+        } 
+        if(langTypeMap.containsKey(Properties.ENGLISH)) {
+        	defaultLang = langTypeMap.get(Properties.ENGLISH);
+        }
+    }
+    
+    public String[] getTessLangs() {
+        String command = "/usr/local/bin/tesseract --list-langs";
+        Process proc;
+        BufferedReader reader;
+        String output = "";
+        String[] lang_list;
+        String[] languages;
+        try {
+			proc = Runtime.getRuntime().exec(command);
+			reader =  new BufferedReader(new InputStreamReader(proc.getInputStream()));
+			String line = "";
+	        while((line = reader.readLine()) != null) {
+	        	output = output + line + " ";
+	        }
+	        proc.waitFor();   
+		} catch (IOException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
+		} catch (InterruptedException e) {
+            messageHandler.handleMessage("Error while getting the Tesserract laguages.", e, MessageType.ERROR);
+		}
+        lang_list = output.split(":");
+        languages = lang_list[1].split(" ");
+        
+        return languages;
```

Comment:
this method does not belong into the controller. there are classes handling tesseract connections. controllers are only meant for transferring data between view and business logic.

Reply:
also, formatting is all messed up.

Reply:
Moved getTessLangs() method to the existing TesseractOCRParser.java class